### PR TITLE
CUDA graph optimizations

### PR DESCRIPTION
**Pull request by phlippe**
Branch for making CUDA graphs runnable. Reference: https://github.com/NX-AI/mlstm_kernels/pull/46

### CODE REVIEW

File: src/transformers/utils/import_utils.py [hunk 1]
```diff
@@ -749,6 +749,7 @@ def is_torch_compile_available():
 def is_torchdynamo_compiling():
     if not is_torch_available():
         return False
+    return True
```
Comment (phlippe) [inline src/transformers/utils/import_utils.py]: this is a quick fix, as this function is missing support for CUDA graph tracing.

**Review by kpoeppel — APPROVED** [review]
Nice work! I think we should keep the improvements on mamba2 / mamba, right?

**Comment by phlippe** [comment]
> I think we should keep the improvements on mamba2 / mamba, right?

Yes, we would need these small changes to allow others to reproduce our benchmarks. Otherwise Mamba/FalconMamba cannot use CUDA Graphs.

**Review by kpoeppel — APPROVED** [review]
Thanks for the nice work!
Regarding the Mamba changes, I think this is even more general and better than the original code. So we should keep it IMO.

**Pull request merged**